Yank/restraints.py

```diff
@@ -1599,33 +1633,33 @@ def determine_missing_parameters(self, thermodynamic_state, sampler_state, topog
         if self._are_restrained_atoms_defined:
             self._determine_restraint_parameters(sampler_state, topography)
         else:
+            # Cache initial selection to reset after each attempt
+            initial_restrained_receptor = self.restrained_receptor_atoms
+            initial_restrained_ligand = self.restrained_ligand_atoms
             # Keep selecting random retrained atoms until the parameters
             # make the standard state correction robust.
             for attempt in range(MAX_ATTEMPTS):
                 logger.debug('Attempt {} / {} at automatically selecting atoms and '
                              'restraint parameters...'.format(attempt, MAX_ATTEMPTS))
 
-                # Randomly pick non-collinear atoms.
+                # Smi-Randomly pick non-collinear atoms with rules
+                self.restrained_ligand_atoms = initial_restrained_ligand
+                self.restrained_receptor_atoms = initial_restrained_receptor
                 restrained_atoms = self._pick_restrained_atoms(sampler_state, topography)
                 self.restrained_receptor_atoms = restrained_atoms[:3]
                 self.restrained_ligand_atoms = restrained_atoms[3:]
 
                 # Determine restraint parameters for these atoms.
                 self._determine_restraint_parameters(sampler_state, topography)
 
+                # TODO: JDC/AXR do we need a check for robustness now that we are on numerical only?
```

Comment:
I realized in my bugfixes that we no longer have the analytical correction, so is this loop where we check robustness even needed anymore?

Reply:
If we're going to use only numerical also for the old functional form, I'm +1 on removing this check.